src/git/repo.rs

```diff
@@ -1231,6 +1432,31 @@ impl<'repo> Commit<'repo> {
             (Some(parent_oid), Some(current_oid)) => Ok(Some(parent_oid != current_oid)),
         }
     }
+
+    /// Amend this existing commit.
+    /// Returns the OID of the resulting new commit.
+    #[instrument]
+    pub fn amend(
```

Comment:
I would call this method `amend_commit` to make it clear what it's operating on.